sys/shell/shell.c

```diff
@@ -225,7 +225,7 @@ static int readline(shell_t *shell, char *buf, size_t size)
         if (c == '\r' || c == '\n') {
             if (line_buf_ptr == buf) {
                 /* The line is empty. */
-                continue;
+                return 1;
             }
 
             *line_buf_ptr = '\0';
```

Comment:
Can you move this line to above the `if` block please?
Without this, pressing enter after a valid command will execute the same command again.